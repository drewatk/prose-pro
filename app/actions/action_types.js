export const FORM_SUBMIT_SUCCEEDED = "@@redux-form/SET_SUBMIT_SUCCEEDED";
export const SELECT_PROJECT = "SELECT_PROJECT";
