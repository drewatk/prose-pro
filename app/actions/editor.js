export const UPDATE_EDITOR_STATE = "UPDATE_EDITOR_STATE";

export const loadFile = file => dispatch => {
  /* TODO: grab file data formatted for Editors */
  // then ...
  console.log(file);
  dispatch({ type: UPDATE_EDITOR_STATE, payload: "TEMP" /* fileData */ });
};
