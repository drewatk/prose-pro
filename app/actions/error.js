export const SHOW_ERROR = "SHOW_ERROR";
export const HIDE_ERROR = "HIDE_ERROR";

export const showError = errorMessage => ({
  type: SHOW_ERROR,
  payload: { errorMessage }
});

export const hideError = () => ({ type: HIDE_ERROR });
