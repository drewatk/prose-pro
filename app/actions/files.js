export const UPDATE_FILES = "UPDATE_FILES";

const updateFiles = files => ({
  type: UPDATE_FILES,
  payload: files
});

export default updateFiles;
