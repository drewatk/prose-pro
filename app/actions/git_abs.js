export const INIT_GIT_ABS = "INIT_GIT_ABS";
export const LOAD_FILE_HISTORY = "LOAD_FILE_HISTORY";

import { openProject } from "app/git-abs";
import { log } from "app/git-abs/git";

const initGitAbs = project => dispatch => {
  console.log(project);
  openProject("app/TestProjects/Project1")
    .then(git_abs => dispatch({ type: INIT_GIT_ABS, payload: git_abs }))
    .catch(err => console.error("Error in initGitAbs action creator: ", err));
};

const loadFileHistory = file => dispatch => {
  console.log(file);
  log(/* file info */)
    .then(commits => dispatch({ type: LOAD_FILE_HISTORY, payload: commits }))
    .catch(err =>
      console.error("Error in loadFileHistory action creator: ", err)
    );
};

export { initGitAbs, loadFileHistory };
