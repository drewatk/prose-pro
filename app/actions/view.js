export const TOGGLE_SHOW_HISTORY = "TOGGLE_SHOW_HISTORY";
export const TOGGLE_SHOW_FILE_LIST = "TOGGLE_SHOW_FILE_LIST";

export function toggleShowHistory() {
  return { type: TOGGLE_SHOW_HISTORY };
}

export function toggleShowFileList() {
  return { type: TOGGLE_SHOW_FILE_LIST };
}
