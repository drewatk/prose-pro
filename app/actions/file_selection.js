import { EditorState, convertFromRaw } from "draft-js";
import { UPDATE_EDITOR_STATE } from "./editor";
import { UPDATE_HISTORY_STATE } from "./history";

export const SELECT_FILE = "SELECT_FILE";

const selectFile = (gitAbs, file) => dispatch => {
  gitAbs
    .openFile(file)
    .then(fileData => {
      dispatch([
        {
          type: SELECT_FILE,
          payload: file
        },
        {
          type: UPDATE_EDITOR_STATE,
          payload: EditorState.createWithContent(convertFromRaw(fileData))
        }
      ]);
      return gitAbs.getVersions(file);
    })
    .then(({ versions }) =>
      dispatch({
        type: UPDATE_HISTORY_STATE,
        payload: versions.map(v => ({
          message: v.getVersionName(),
          commitHash: v.getCommitId(),
          date: v.getTimestamp()
        }))
      })
    )
    .catch(err => console.error("Error in file select action creator: ", err));
};

export default selectFile;
