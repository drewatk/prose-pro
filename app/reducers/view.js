import { TOGGLE_SHOW_HISTORY, TOGGLE_SHOW_FILE_LIST } from "../actions/view";

const initialState = {
  showHistory: true,
  showFileList: true
};

export default function editor(state = initialState, { type }) {
  switch (type) {
    case TOGGLE_SHOW_HISTORY:
      return Object.assign({}, state, {
        showHistory: !state.showHistory
      });

    case TOGGLE_SHOW_FILE_LIST:
      return Object.assign({}, state, {
        showFileList: !state.showFileList
      });
  }
  return state;
}
