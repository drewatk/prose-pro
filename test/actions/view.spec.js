import { toggleShowHistory, toggleShowFileList } from "app/actions/view";

describe("View Actions", () => {
  test("Toggle Show History", () => {
    expect(toggleShowHistory()).toMatchSnapshot();
  });
  test("Toggle Show File List", () => {
    expect(toggleShowFileList()).toMatchSnapshot();
  });
});
