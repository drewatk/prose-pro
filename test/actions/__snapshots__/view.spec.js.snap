// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Actions Toggle Show File List 1`] = `
Object {
  "type": "TOGGLE_SHOW_FILE_LIST",
}
`;

exports[`View Actions Toggle Show History 1`] = `
Object {
  "type": "TOGGLE_SHOW_HISTORY",
}
`;
